app/services/actual/import.rb

```diff
@@ -155,12 +160,50 @@ def to_h
       end
 
       def convert_to_attributes
+        validate_values
         attrs = {value: convert_values}
         NON_VALUE_FIELDS.each_with_object(attrs) do |(attr_name, column_name), attrs|
           attrs[attr_name] = convert_to_attribute(attr_name, @row[column_name])
         end
       end
 
+      def validate_values
+        values_string = [
+          @row["Actual Value"] || "blank",
+          @row["Refund Value"] || "blank"
+        ].join(", ")
+
+        types = {
+          actual: value_type(value: @row["Actual Value"], field: :actual_value),
+          refund: value_type(value: @row["Refund Value"], field: :refund_value)
+        }
+
+        error_message = case types
+        when {actual: :non_zero, refund: :non_zero}
```

Comment:
I like this! It's a nice way of avoiding a bunch of nested `if`s and feels easy to extend for future errors.

Reply:
Yeah I was trying for a while to work out a clean way of not nesting too much and making things readable, while also dealing with the complicated stuff like financial data conversion. It was all a little messy